packages/core-flows/src/definition/cart/workflows/add-shipping-method-to-cart.ts

```diff
@@ -0,0 +1,75 @@
+import { CartDTO } from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common/steps/use-remote-query"
+import { addShippingMethodToCartStep } from "../steps"
+import { getShippingOptionPriceSetsStep } from "../steps/get-shipping-option-price-sets"
+import { refreshCartPromotionsStep } from "../steps/refresh-cart-promotions"
+
+interface AddShippingMethodToCartWorkflowInput {
+  cart: CartDTO
```

Comment:
suggestion: I believe that the input should only expect what they need to work, here it looks like the id and currency code would be enough so I would say that it is not necessary to enforce the rest of the data to be provided wdyt? It should be always the approach for all workflows, always asking what they need not all the data.